packages/smooth_app/lib/pages/product/nutrition_page_loaded.dart

```diff
@@ -559,9 +562,20 @@ class _NutritionPageLoadedState extends State<NutritionPageLoaded> {
     //And if it is not, we create a new product with the fields of the _product
     // and we insert it in the database. (Giving the user an immediate feedback)
     if (product != null) {
+      product.noNutritionData = changedProduct.noNutritionData;
```

Comment:
Again, please create a private method that populate the product fields, and call it twice.